APE14.rst

```diff
@@ -406,6 +426,74 @@ dimension encoding time-of-observation.
+and we therefore provide two methods for each transformation.
+
+We do not mandate a specific order for the world coordinates. While it might be
+tempting to assume the 'same' order as for pixel coordinates, this only makes
+sense for simple cases (for example an image of the sky where ra/dec are roughly
+lined up with x/y). In a generalized WCS system, such a correspondance does not
+exist. As an example, consider an equatorial coordinate system rotated 45
+degrees from the pixel coordinates. Such a system could be represented by the
+following::
+
+    wcs.axis_correlation_matrix = [[True, True], [True, True]]
+    wcs.world_axis_physical_types = ['pos.eq.ra', 'pos.eq.dec']
+
+or by::
+
+    wcs.axis_correlation_matrix = [[True, True], [True, True]]
+    wcs.world_axis_physical_types = ['pos.eq.dec', 'pos.eq.ra']
+
+Neither of these is more correct than the other since ra/dec are not
+preferentially lined up with x/y, so we need to allow both.
+
+It is also possible to have a different number of world coordinates compared to
+pixel coordinates. For example, we could imagine having a 1D array of values
+determined by tracing a non-linear path through a spectral cube. The WCS would
+look like::
+
+    wcs.axis_correlation_matrix = [[True], [True], [True]]
+    wcs.world_axis_physical_types = ['pos.eq.ra', 'pos.eq.dec', 'spect.dopplerVeloc.radio']
+
+but the order of the coordinates is of course arbitrary, and one cannot simply
+refer to the order of the pixel coordinates since there is only one pixel
+coordinate that is correlated with all three world coordinates. Thus, one could
+equally represent the WCS as:
+
+    wcs.axis_correlation_matrix = [[True], [True], [True]]
+    wcs.world_axis_physical_types = ['spect.dopplerVeloc.radio', 'pos.eq.ra', 'pos.eq.dec']
+
+and there is no 'right' order.
+
+We note that the API we present here makes it easy to create a WCS with
+reorederd world coordinates - this would involve changing the order of
+``world_axis_physical_types``, ``world_axis_units``, and
+``world_axis_object_components``, changing the order of
+``axis_correlation_matrix`` along the first dimension, and changing the order
+of the inputs of the ``world_to_pixel/index_values`` methods and the
+order of the outputs of the ``pixel/index_to_world_values`` methods. Thus,
+implementations of the low or high-level API could provide convenience methods
+to reorder or sort the world axes.
+
+This flexibility does not however extend to pixel coordinates. For example for a
+given array with an associated WCS, the output of ``world_to_index_values`` has
+to consistently return the values in the order that can be used to index the
+array, so the indices/pixel coordinates of the WCS cannot be re-ordered if the
+data is left unchanged.
+
+For consistency with existing WCS libraries, we recommend that implementations
+based on FITS-WCS choose to order ``world_axis_physical_types`` in the same
+order as the ``CDELT`` values, but we do not require this.
```

Comment:
Did you mean `CTYPE` here?

Reply:
woops yes

Reply:
Fixed in 5c66cedf12db349f306f50f3b823212427be5a94